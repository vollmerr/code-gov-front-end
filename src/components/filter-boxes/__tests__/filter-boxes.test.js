import React from 'react';

import FilterBoxes from 'components/filter-boxes';
import FilterBox from 'components/filter-box';

import { shallow } from 'enzyme';

describe('Testing <FilterBoxes /> component', () => {
  
  const props = {
    config: [
      ['Language', 'languages'],
      ['State Agency', 'agencies'],
      ['Licenses', 'licenses'],
      ['Usage Types', 'usageTypes']
    ],
    boxes: {
      agencies: [{name: 'Example Dept', value: 'Exa', checked: false}],
      languages: [{name: 'Example Language', value: 'ex', checked: false}],
      licenses: [{name: 'Example license', value: 'Example-License', checked: false}],
      usageTypes: [{name: 'Example Useage', value: 'Example-Useage', checked: false}]
    },
    onChange: jest.fn() 
  };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FilterBoxes {...props} />);
  });
  it('renders all four filter boxes', () => {
    expect(wrapper.find(FilterBox)).toHaveLength(4);
  });
  it('updates when changing props', () => {
    const shouldUpdate = wrapper.instance().shouldComponentUpdate('nextProps', 'nextState');
    expect(shouldUpdate).toBe(true);
  });
  it('component is null when not given props', () => {
    wrapper = shallow(<FilterBoxes />);
    expect(wrapper.getElement()).toBe(null);
  });
  it('should render correct', () => {
    expect(wrapper).toMatchSnapshot();
  })
})