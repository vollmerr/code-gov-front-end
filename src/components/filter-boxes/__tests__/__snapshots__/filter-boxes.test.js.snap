// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FilterBoxes /> component should render correct 1`] = `
<Fragment>
  <FilterBox
    key="Language"
    onChange={[Function]}
    options={
      Array [
        Object {
          "checked": false,
          "name": "Example Language",
          "value": "ex",
        },
      ]
    }
    title="Language"
  />
  <FilterBox
    key="State Agency"
    onChange={[Function]}
    options={
      Array [
        Object {
          "checked": false,
          "name": "Example Dept",
          "value": "Exa",
        },
      ]
    }
    title="State Agency"
  />
  <FilterBox
    key="Licenses"
    onChange={[Function]}
    options={
      Array [
        Object {
          "checked": false,
          "name": "Example license",
          "value": "Example-License",
        },
      ]
    }
    title="Licenses"
  />
  <FilterBox
    key="Usage Types"
    onChange={[Function]}
    options={
      Array [
        Object {
          "checked": false,
          "name": "Example Useage",
          "value": "Example-Useage",
        },
      ]
    }
    title="Usage Types"
  />
</Fragment>
`;
